packages/dd-trace/src/opentelemetry/metrics/instruments.js

```diff
@@ -0,0 +1,147 @@
+'use strict'
+
+const { sanitizeAttributes } = require('@opentelemetry/core')
+const { METRIC_TYPES } = require('./constants')
+
+/**
+ * @typedef {import('@opentelemetry/api').Attributes} Attributes
+ * @typedef {import('@opentelemetry/core').InstrumentationScope} InstrumentationScope
+ */
+
+/**
+ * Base class for all metric instruments.
+ *
+ * This implementation follows the OpenTelemetry JavaScript API Instrument interface:
+ * https://open-telemetry.github.io/opentelemetry-js/interfaces/_opentelemetry_api._opentelemetry_api.Instrument.html
+ * @private
+ */
+class Instrument {
+  #baseMetadata
+
+  constructor (name, options, instrumentationScope, reader) {
+    this.name = name
+    this.description = options.description || ''
+    this.unit = options.unit || ''
+    this.instrumentationScope = instrumentationScope
+    this.reader = reader
+    // Pre-create static measurement fields to avoid object creation on every measurement
+    this.#baseMetadata = {
+      name: this.name,
+      description: this.description,
+      unit: this.unit,
+      instrumentationScope: this.instrumentationScope
+    }
+  }
+
+  createMeasurement (type, value, attributes) {
+    return {
+      ...this.#baseMetadata,
```

Comment:
It should actually be faster or as fast doing the assignment directly.

```suggestion
  }

  createMeasurement (type, value, attributes) {
    return {
      name: this.name,
      description: this.description,
      unit: this.unit,
      instrumentationScope: this.instrumentationScope,
```